server/api/rounds.py

```diff
@@ -196,28 +198,141 @@ def cumulative_batch_results(election: Election) -> BatchTallies:
     }
 
 
+def cvrs_for_contest(contest: Contest) -> supersimple.CVRS:
+    choice_name_to_id = {choice.name: choice.id for choice in contest.choices}
+
+    cvrs: supersimple.CVRS = defaultdict(lambda: {contest.id: {}})
+
+    for jurisdiction in contest.jurisdictions:
+        cvr_contests_metadata = typing_cast(
+            JSONDict, jurisdiction.cvr_contests_metadata
+        )
+        choices_metadata = cvr_contests_metadata[contest.name]["choices"]
+
+        interpretations_query = (
+            CvrBallot.query.join(Batch)
+            .filter_by(jurisdiction_id=jurisdiction.id)
+            .join(
+                SampledBallot,
+                and_(
+                    CvrBallot.batch_id == SampledBallot.batch_id,
+                    CvrBallot.ballot_position == SampledBallot.ballot_position,
+                ),
+            )
+        )
+        # For targeted contests, use the ticket number to key the ballots so
+        # that we count all sample draws
+        if contest.is_targeted:
+            interpretations_by_ballot = (
+                interpretations_query.join(SampledBallotDraw)
+                .filter(SampledBallotDraw.contest_id == contest.id)
+                .values(SampledBallotDraw.ticket_number, CvrBallot.interpretations)
+            )
+        # For opportunistic contests, use the ballot id to key the ballots so
+        # that we only count unique ballots
+        else:
+            interpretations_by_ballot = interpretations_query.values(
+                SampledBallot.id, CvrBallot.interpretations
+            )
+
+        for ballot_key, interpretations_str in interpretations_by_ballot:
+            # interpretations is the raw CVR string: 1,0,0,1,0,1,0. We need to
+            # pick out the interpretation for each contest choice. We saved the
+            # column index for each choice when we parsed the CVR.
+            interpretations = interpretations_str.split(",")
+            for choice_name, choice_metadata in choices_metadata.items():
+                interpretation = interpretations[choice_metadata["column"]]
+                # If the interpretations are empty, it means the contest wasn't
+                # on the ballot, so we should skip this contest entirely for
+                # this ballot.
+                if interpretation == "":
+                    cvrs[ballot_key] = {}
```

Comment:
do you mean to zero the whole CVR, not just the value for `contest.id` ?

Reply:
this is what the audit math expects /shrug

Reply:
ohhh wait i see what you're saying

Reply:
oh no wait it does work because there's only ever one contest (because the function - and the audit math) operates on a contest by contest basis